lib/TestScriptRunnable.rb

```diff
@@ -168,31 +173,32 @@ def evaluate assertion
     report.pass
   end
   
-  def extract_path op
-    return replace_variables(op.url) if op.url
-    pieces = { format: FORMAT_MAP[op.contentType] }
-
-    if op.targetId
-      pieces[:id] = id_map[op.targetId]
-      pieces[:resource] = find_resource(op.targetId).resourceType
-
-      throw :exit, report.fail('noId') unless pieces[:id]
-      throw :exit, report.fail('noTargetIdFixture') unless pieces[:resource]
-    elsif op.params
-      pieces[:resource] = replace_variables op.resource
-      pieces[:params] = replace_variables op.params
-
-      throw :exit, report.fail('noResource') unless pieces[:resource]
-    elsif op.sourceId
-      pieces[:resource] = find_resource(op.sourceId).resourceType
-
-      throw :exit, report.fail('noSourceFixture') unless fixtures[op.sourceId].resourceType
+  def extract_path(operation, request_type)
+    return replace_variables(operation.url) if operation.url
+
+    if operation.params
+      return if operation.resource.nil? && requires_type(operation)
+
+      mime = "{&_format=#{FORMAT_MAP[operation.contentType]}}" if operation.contentType
+      params = "#{replace_variables(operation.params)}#{mime}"
+      search = '/_search' if request_type == :post
+      "#{operation.resource}#{search}#{params}"
+    elsif operation.targetId
+      type = response_map[operation.targetId]&.resource.resourceType
+      id = id_map[operation.targetId]
+      return "#{type}/#{id}" unless type.nil? || id.nil?
+    elsif operation.sourceId
+      fixtures[operation.sourceId]&.resourceType
     end
+  end
 
-    # TODO: requestEncodeUrl?
-    client.resource_url(pieces)
+  # Determines if the operation requires [type] as part of
+  # its intended request url
+  def requires_type(operation)
```

Comment:
Not sure how necessary this is as its own method

Reply:
In the sense of extract_path() is only used by execute(), probably no. But putting this directly into execution would make it quite nasty. I'd vote to keep as separate for now.